gnss_to_map_convertor/src/gnss_to_map_convertor/GNSSToMapNode.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2018-2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <gnss_to_map_convertor/GNSSToMapNode.h>
+#include <tf2_ros/transform_listener.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2/LinearMath/Matrix3x3.h>
+
+GNSSToMapNode::GNSSToMapNode() : tfListener_(tfBuffer_), p_cnh_("~") {}
+
+void GNSSToMapNode::fixCb(const novatel_gps_msgs::NovatelPositionConstPtr& fix_msg, const novatel_gps_msgs::NovatelDualAntennaHeadingConstPtr& heading_msg) {
+  // Get sensor and base_link transforms if not yet loaded
+  if (!baselink_in_sensor_set_) {
+    try {
+      tf2::convert(tfBuffer_.lookupTransform(fix_msg->header.frame_id, base_link_frame_, ros::Time(0)).transform, baselink_in_sensor_);
+      tf2::convert(tfBuffer_.lookupTransform(ned_heading_frame_, fix_msg->header.frame_id, ros::Time(0)).transform, sensor_in_ned_);
+    } catch (tf2::TransformException &ex) {
+      ROS_WARN_STREAM("Ignoring fix message: Could not locate static transforms with exception " << ex.what());
+      return;
+    }
+    baselink_in_sensor_set_ = true;
+  }
+
+  sensor_msgs::NavSatFix nav_fix_msg;
+  nav_fix_msg.latitude = fix_msg->lat;
+  nav_fix_msg.longitude = fix_msg->lon;
+  nav_fix_msg.altitude = fix_msg->height;
+  sensor_msgs::NavSatFixConstPtr fix_ptr(new sensor_msgs::NavSatFix(nav_fix_msg));
+
+  geometry_msgs::PoseWithCovarianceStamped ecef_pose = gnss_to_map_convertor::poseFromGnss(
+    baselink_in_sensor_, sensor_in_ned_, fix_ptr, heading_msg
+  ); 
+  ecef_pose.header.frame_id = earth_frame_; // Set correct frame id
```

Comment:
Do we need to set header for ecef_pose?

Reply:
The header is assigned inside the poseFromGnss function

Reply:
Got you.